### PR TITLE
Setting up MARL Benchmark with SMARTS 0.6.1

### PR DESCRIPTION
**Pull request by RutvikGupta**
Restructured `baselines/marl_benchmark` folder to allow setting up with `smarts==0.6.1` and updated package requirements in `setup.py` and `README.md` to resolve training procedure errors.

### CODE REVIEW

**Comment by RutvikGupta** [comment]
The changes mentioned by @zyk516 in  Issue #1131 have been made here. Now one can follow the instructions of `README.md` and get the training and evaluate procedures to run for MARL benchmark.

**Comment by Gamenot** [comment]
This should now be unblocked since #1138 is merged.

**Comment by Gamenot** [comment]
It looks like https://github.com/huawei-noah/SMARTS/issues/1230 will possibly block this.

**Comment by Gamenot** [comment]
https://github.com/huawei-noah/SMARTS/pull/1235 has merged in to address a secondary exception can block training.

**Comment by RutvikGupta** [comment]
> It looks like #1230 will possibly block this
I ran the training with social agent scenarios and saw that it is not crashing. So maybe it won't block this pr.

**Comment by RutvikGupta** [comment]
> #1235 has merged in to address a secondary exception can block training.

Would need a new release to allow using it for this pr. Converting the PR to draft for now.

**Comment by RutvikGupta** [comment]
The PR is blocked by the unresolved ray issue: https://github.com/ray-project/ray/issues/5339. 
Need to resolve the above issue to support multi-agent training using several policies with checkpoint learning (or export models). Or need to remove the support to export checkpoint and model. More discussion is required.

**Comment by Gamenot** [comment]
Checkpoints are magically working for some reason on my environment...

File: baselines/marl_benchmark/README.md [hunk 1]
```diff
@@ -3,19 +3,19 @@
...
-- `evaluate.py`: The evaluation program
-- `run.py`: Executes multi-agent training
+- `marl_benchmark/agents/`: YAML files and some RLlib-based policy implementations
+- `marl_benchmark/metrics/`: Class definition of metrics (default by a basic Metric class)
+- `marl_benchmark/networks/`: Custom network implementations
+  - `marl_benchmark/communicate.py`: Used for Networked agent learning
+- `marl_benchmark/scenarios/`: Contains three types of scenarios tested in the paper
+- `marl_benchmark/wrappers/`: Environment wrappers
+- `marl_benchmark/evaluate.py`: The evaluation program
+- `marl_benchmark/run.py`: Executes multi-agent training
```
Comment (saulfield) [inline baselines/marl_benchmark/README.md:13]: I feel like prepending the new folder name just adds noise to the documentation.

**Review by saulfield — APPROVED** [review]
Changes look fine to me. I'm just wondering why the directory restructuring was needed. Would relative imports accomplish the same thing?

**Comment by Gamenot** [comment]
@saulfield 
> Changes look fine to me. I'm just wondering why the directory restructuring was needed. Would relative imports accomplish the same thing?

I am not entirely certain why the directory was restructured. However, it does make the benchmark obviously packageable.

**Pull request closed without merging**